### PR TITLE
[feature] Added a CLI flag to allow passing without any test files found

### LINKED ISSUE #1331: Allow to pass if no tests
If I run e.g. `web-test-runner "src/**/*.test.tsx"` and there are no files matching that pattern, I get this error:
```
Error: Could not find any test files with pattern(s): src/**/*.test.tsx
```

Would be nice if there was a flag that could be passed to allow it to pass if no files are found. This is useful for scaffolds and other initial stages.

### PR DESCRIPTION
**Pull request by glektarssza**
## What I did

1. Added a CLI flag that allows passing without any test files being found.
2. Added documentation for the above new CLI flag.
3. Added a Git ignore line for local environment files (e.g. `.envrc`).

Would resolve #1331

### CODE REVIEW

**Comment by Westbrook** [comment]
Could you add a change set as [requested above](https://github.com/modernweb-dev/web/pull/2758#issuecomment-2183220312)?

**Comment by glektarssza** [comment]
I will add one after work today.

**Comment by glektarssza** [comment]
Changeset added, @Westbrook. I think the bot's comment is completely out to lunch now, though, because of the merge commit in a957ff9.

**Comment by Westbrook** [comment]
Thanks for getting the changeset it. That is looking fine. Can you check in the CI output? Once that’s green we should be good to edge this.

**Comment by glektarssza** [comment]
Will take a look tonight!

File: packages/test-runner/src/config/parseConfig.ts [hunk 1]
```diff
@@ -188,8 +188,10 @@ export async function parseConfig(
       // we can improve this by relying only on groups inside the test runner itself
       if (groupConfig.files == null) {
         groupConfig.files = finalConfig.files;
+        groupConfig.allowPassWithoutTests = finalConfig.allowPassWithoutTests;
       }
       finalConfig.files = undefined;
+      finalConfig.allowPassWithoutTests = undefined;
```
Comment (bashmish) [inline packages/test-runner/src/config/parseConfig.ts:194]: what is the reason to set it as `undefined`? isn't it more clear to have it always as a boolean and default to `false`?

File: packages/test-runner-core/src/config/TestRunnerCoreConfig.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ export interface CoverageConfig {
 export interface TestRunnerCoreConfig {
   rootDir: string;
   files?: string | string[];
+  allowPassWithoutTests?: boolean;
```
Comment (bashmish) [inline packages/test-runner-core/src/config/TestRunnerCoreConfig.ts:30]: related to another question: I'd just make it `allowPassWithoutTests: boolean` and default to "false", here and below

**Pull request closed without merging**